Sources/PackageCollections/PackageCollections.swift

```diff
@@ -530,17 +544,17 @@ public struct PackageCollections: PackageCollectionsProtocol {
                     collectionIdentifiers = collectionIdentifiers.filter { collections.contains($0) }
                 }
                 if collectionIdentifiers.isEmpty {
-                    return callback(.failure(NotFoundError("\(reference)")))
+                    return callback(.failure(NotFoundError("\(identity)")))
                 }
-                self.storage.collections.findPackage(identifier: reference.identity, collectionIdentifiers: collectionIdentifiers) { findPackageResult in
+                self.storage.collections.findPackage(identifier: identity, collectionIdentifiers: collectionIdentifiers) { findPackageResult in
                     switch findPackageResult {
                     case .failure(let error):
                         callback(.failure(error))
                     case .success(let packagesCollections):
                         // A package identity can be associated with multiple repository URLs
-                        let matches = packagesCollections.packages.filter { $0.reference.canonicalLocation == reference.canonicalLocation }
+                        let matches = packagesCollections.packages.filter { $0.identity == identity }
```

Comment:
I think this filtering is no longer needed because `$0.identity == identity` should be `true` for all the items in the result.

If we are confident that identity offers "enough" uniqueness, we can potentially revert some of the changes we made in https://github.com/apple/swift-package-manager/pull/3543, but at the minimum we can remove this `filter`.

Reply:
oh interesting, I missed that point. we could also add the location as an argument here and continue to do the canonical check. wdyt?

Reply:
but in general... the idea of identity is that its unique, so if two collections offer different packages with the same identity, there is def something a bit wrong going on

Reply:
> the idea of identity is that its unique, so if two collections offer different packages with the same identity, there is def something a bit wrong going on

Right, that was the original assumption. 

> we could also add the location as an argument here and continue to do the canonical check. wdyt?

Yes, I think I would be more comfortable with this until we completely switch over to identity.

Reply:
@yim-lee https://github.com/apple/swift-package-manager/pull/3743/commits/aae361d7a97f7cba6a065f93e6577e3d630b0aa8

wdyt?